lib/jsonapi/view.ex

```diff
@@ -104,6 +113,12 @@ defmodule JSONAPI.View do
         def type, do: raise("Need to implement type/0")
       end
 
+      if @namespace do
```

Comment:
This felt really dirty to me, though I have no real good reason to feel that way. Is there a better solution?

Reply:
I'm not entirely sure if there's an "ideal" way to handle this. We could skip the attribute and call `unquote(namespace)` directly?

```elixir
      if unquote(namespace) do
        def namespace, do: unquote(namespace)
      else
        def namespace, do: Application.get_env(:jsonapi, :namespace, "")
      end
```

Or maybe we just make it simple?


```elixir
def namespace, do: unquote(namespace) || Application.get_env(:jsonapi, :namespace, "")
```

Reply:
> Or maybe we just make it simple? ...

This version resulted in some weird warnings:

```elixir
warning: this check/guard will always yield the same result
  test/jsonapi/view_test.exs:5

warning: this check/guard will always yield the same result
  test/jsonapi/view_test.exs:19

warning: this check/guard will always yield the same result
  test/jsonapi/view_test.exs:43
```

These all point to the `use JSONAPI.View, namespace: "/foo"` invocations in our test views. I don't quite understand what's happening with that.

That said your first option worked, but didn't seem much better than what I had already. I've left my original solution in place. I'm more than open to changing it though.

Reply:
What you have is fine 😁